plugin/flags_sources/compilation_db.py

```diff
@@ -200,9 +200,13 @@ def filter_bad_arguments(argument_list):
                     skip_next = True
 
                 continue
-            if i == len(argument_list) - 1:
-                # ignore the last element as it is a file to compile, something
-                # like 'test.cpp'
+            if argument.endswith('.c'):
```

Comment:
This is not enough as the extensions can be arbitrary. For example, Google tends to use `.cc`, etc.

Reply:
We can list all possible extensions or compare argument with file name/path.

Reply:
But why is the last element not the file to compile? What is the last element in such a line?

Reply:
Last element is output file, right after file to compile.